pages/scenarios/[runId].vue

```diff
@@ -120,56 +121,82 @@ import type { Parameter } from "~/types/parameterTypes";
+
 const showSpinner = computed(() => {
   return (!appStore.currentScenario.result.data
-    && appStore.currentScenario.result.fetchStatus !== "error"
-    && !stoppedPolling.value);
+    && appStore.currentScenario.result.fetchStatus !== "error");
 });
 
 const paramDisplayText = (param: Parameter) => {
   if (appStore.currentScenario?.parameters && appStore.currentScenario?.parameters[param.id]) {
     const rawVal = appStore.currentScenario.parameters[param.id].toString();
+
+    const rawValIsNumberString = Number.parseInt(rawVal).toString() === rawVal;
+    if (rawValIsNumberString) {
+      // TODO: Localize number formatting.
+      return new Intl.NumberFormat().format(Number.parseInt(rawVal));
+    }
     return param.options ? param.options.find(({ id }) => id === rawVal)!.label : rawVal;
   }
 };
 
+// Use useAsyncData to store the time once, during server-side rendering: avoids client render re-writing value.
+const { data: timeOfFirstStatusPoll } = await useAsyncData<number>("timeOfFirstStatusPoll", async () => {
+  return new Promise<number>((resolve) => {
+    resolve(new Date().getTime());
+  });
+});
+
 // Eagerly try to load the status and results, in case they are already available and can be used during server-side rendering.
 await appStore.loadScenarioStatus();
 if (appStore.currentScenario.status.data?.runSuccess) {
   appStore.loadScenarioResult();
 }
 
-let statusInterval: NodeJS.Timeout;
-const loadScenarioStatus = () => {
-  appStore.loadScenarioStatus().then(() => {
-    if (appStore.currentScenario.status.data?.runSuccess) {
-      clearInterval(statusInterval);
-      jobTakingLongTime.value = false;
-      appStore.loadScenarioResult();
-    }
-  });
+watch(() => appStore.currentScenario.status.data?.runSuccess, (runSuccess) => {
+  if (runSuccess) {
+    appStore.loadScenarioResult();
+  }
+});
+
+watch(() => appStore.currentScenario.status.data?.done, (done) => {
+  if (done) {
+    clearInterval(statusInterval);
+    jobSlow.value = false;
+    jobReallySlow.value = false;
+  }
+});
+
+const pollForStatusEveryNSeconds = (seconds: number) => {
+  statusInterval = setInterval(() => {
+    if (timeOfFirstStatusPoll.value) {
+      secondsSinceFirstStatusPoll.value = ((new Date().getTime() - timeOfFirstStatusPoll.value) / 1000).toFixed(0);
+    };
+    appStore.loadScenarioStatus();
+  }, seconds * 1000);
 };
 
 onMounted(() => {
   appStore.globe.interactive = false;
 
   if (!appStore.currentScenario.status.data?.done && appStore.currentScenario.runId) {
-    statusInterval = setInterval(loadScenarioStatus, 200); // Poll for status every N ms
+    pollForStatusEveryNSeconds(0.2);
     setTimeout(() => {
-      // If the job isn't completed within five seconds, give user the information about the run status.
+      // If the job isn't completed within M seconds, give user the information about the run status.
       if (appStore.currentScenario.status.data?.runStatus !== runStatus.Complete) {
```

Comment:
Not from this branch, but I just noticed that the two timeouts use different ways to tell if a job is finished: 
`appStore.currentScenario.status.data?.runStatus !== runStatus.Complete` vs `!appStore.currentScenario.status.data?.done` - I think the latter is more reliable because status can be in error, but the job is still done.